python/res/enkf/jobs/enkf_scaling_job.py

```diff
@@ -0,0 +1,220 @@
+    wildcard_key = {"key": "WOPT*"}
+    """
+    if "index" in wildcard_key:
+        return [{"key": key, "index": wildcard_key["index"]} for key in matching_keys]
+    else:
+        return [{"key": key} for key in matching_keys]
+
+
+def _expand_wildcard(ert, wildcard_key, parent):
+    """
+    Expands a wildcard, the parent will be different if it is a calculate key
+    or update key, so a different approach is needed to expand them.
+    """
+    matching_keys = ert.getObservations().getMatchingKeys(wildcard_key).strings
+    if isinstance(parent, dict):
+        return _wildcard_to_dict_list(matching_keys, wildcard_key)
+    elif isinstance(parent, str):
+        return [key for key in matching_keys]
+
+
+def _find_and_expand_wildcards(ert, user_dict):
+    """
+    Loops through the user input and identifies wildcards in observation
+    names and expands them.
+    """
+    new_dict = deepcopy(user_dict)
+    for main_key, value in user_dict.items():
+        new_entries = []
+        if main_key in ("UPDATE_KEYS", "CALCULATE_KEYS"):
+            path = "key"
+            if main_key == "CALCULATE_KEYS":
+                path = slice(None)
+            for val in value["keys"]:
+                key = val[path]
+                if "*" in key:
+                    new_entries.extend(_expand_wildcard(ert, key, val))
+                else:
+                    new_entries.append(val)
+            new_dict[main_key]["keys"] = new_entries
+
+    return new_dict
+
+
+def setup_configuration(input_data, schema):
+    """
+    Creates a ConfigSuite instance and inserts default values
+    """
+    default_layer = enkf_scaling_job_config.get_default_values()
+    config = configsuite.ConfigSuite(input_data, schema, layers=(default_layer,))
+    return config
+
+
+def _create_active_lists(enkf_observations, events):
+    """
+    Will add observation vectors to observation data. Returns
+    a list of tuples mirroring the user config but also containing
+    the active list where the scaling factor will be applied.
+    """
+    new_events = []
+    observation_data = LocalObsdata("some_name", enkf_observations)
+    for event in events:
+        observation_data.addObsVector(enkf_observations[event.key])
+
+        new_active_list = _get_active_list(observation_data, event.key, event.index)
+
+        new_events.append(_make_tuple(event.key, event.index, new_active_list))
+
+    return new_events
+
+
+def _get_active_list(observation_data, key, index_list):
+    """
+    If the user doesn't supply an index list, the existing active
+    list from the observation is used, otherwise an active list is
+    created from the index list.
+    """
+    if index_list is not None:
+        return _active_list_from_index_list(index_list)
+    else:
+        return observation_data.copy_active_list(key).setParent()
+
+
+def _make_tuple(key, index, active_list):
+    new_event = namedtuple("named_dict", ["key", "index", "active_list"])
```

Comment:
`namedtuple` creates a lightweight class, and should probably not be redefined every time you call `_make_tuple`

Reply:
:+1: